### DISCUSSION TITLE
LAMMPS failed with C10::Error

### DISCUSSION THREAD

**Discussion by Geonu-AI** [mir-group/pair_nequip#25] [Q&A]
Hi, 

Now I am trying to do inference in lammps with my trained model. Here is a part of my error. 
```
**terminate called after throwing an instance of 'c10::Error'
what():  expected scalar type Float but found Byte
frame #0: c10::Error::Error(c10::SourceLocation, std::string) + 0x42 (0x7f02b84437d2 in /my_env/lib/python3.8/site-packages/torch/lib/libc10.so)
frame #1: c10::detail::torchCheckFail(char const*, char const*, unsigned int, std::string const&) + 0x5b (0x7f02b843fe6b in /my_env/lib/python3.8/site-packages/torch/lib/libc10.so)
frame #2: float* at::TensorBase::data_ptr<float>() const + 0xde (0x7f02f568d54e in /my_env/lib/python3.8/site-packages/torch/lib/libtorch_cpu.so)
frame #3: at::TensorAccessor<float, 2ul, at::DefaultPtrTraits, long> at::TensorBase::accessor<float, 2ul>() const & + 0xcb (0x8ecc8b in /lammps-stable_29Sep2021/build/lmp)
.... **
```
It looks similar to a previous discussion ("LAMMPS failed with model trained with float64") but my model was trained with float32, so may not be related.... Could you give me a comment the error ?

**Comment by Linux-cpp-lisp** [comment]
Hi @Geonu-AI ,

This is a very strange error--- did you try to do anything unusual while training your model? Mixed precision training or something?

Can you try upgrading to the latest stable PyTorch (1.11 currently)? This often fixes mysterious incomprehensible errors.
